src/vote.rs

```diff
@@ -98,39 +120,45 @@ impl<T: Proposition> VoteCount<T> {
         faulty: &BTreeSet<NodeId>,
     ) -> Self {
         let mut count: VoteCount<T> = VoteCount::default();
-        let mut counted: BTreeSet<SignatureShare> = Default::default();
+
+        let mut votes_by_honest_voter: BTreeMap<NodeId, SignedVote<T>> = Default::default();
```

Comment:
Quickcheck did end up finding a scenario where the old "count all votes including superseded votes"  approach fails to terminate. If We remove old votes from the set and only consider the latest votes, we get a decision in those scenarios.